File: src/communication/Communication.cpp
#include "communication/Communication.hpp"
#include "Logs/EasyLogging.h"

Communication::Communication(GameController* gamecontroller, TeamController* teamController)
    : gameController(gamecontroller), teamController(teamController)
{
    isRunning = true;
    gameControllerIsConnected = gameController->verifyConnection();
    teamControllerIsConnected = teamController->verifyConnection();
    teamControllerIsConnected = false;
    if (!gameControllerIsConnected) delete gameController;
    if (!teamControllerIsConnected) delete teamController;
}

void Communication::close() {
    LOG(INFO) << "\x1B[93m[COMMUNICATION] Closing controllers\x1B[0m";
    isRunning = false;
    sleep(1);
    if (gameControllerIsConnected) delete gameController;
    if (teamControllerIsConnected) delete teamController;
}

void Communication::process() {
    if (!gameControllerIsConnected && !teamControllerIsConnected) return;
    while (isRunning) {
        sleep(FREQUENCY);
        if (gameControllerIsConnected) gameController->process();
        if (teamControllerIsConnected) teamController->process();
    }
}

File: Communication/communication.py
from Libs.Logs import logs
import time

class Communication:
    def __init__(self, session):
        self.session = session
        self.isRunning = True

    def run(self):
        logs.logger_communication.info("Running Communication")
        while self.isRunning:
            self.__process()
            time.sleep(0.001)

    def close(self):
        logs.logger_communication.info("Closing Communication")
        self.isRunning = False

    def __process(self):
        logs.logger_communication.info("code here")